### PR TITLE
[entropy_src/rtl] remove AsyncOn enables

### PR DESCRIPTION
**Pull request by mwbranstad**
The AsyncOn option was set for some of the prim_mubi4/8_sync blocks.
There are no async clock crossing in this module, so all set to zero.

Signed-off-by: Mark Branstad <mark.branstad@wdc.com>

### CODE REVIEW

**Review by tjaychen — APPROVED** [review]

**Review by martin-lueker — CHANGES_REQUESTED** [review]
Hi @mwbranstad,
   Have you tested this with the RNG test?  I suspect this isn't going to work.  This block is using the delays here for a number of reasons isn't it?  For instance, the health test clear pulse compares the "Async" vs. the non-async enables.   
   It may seem harsh, but I'm going to put a hold on this this until we figure out what we really want here, as I suspect the `es_enable_q_fo` pulses need at least a one-cycle delay, if not two. In fact in parallel a PR I've added another explicit flop to create a 3-cycle delay.

**Comment by martin-lueker** [comment]
So @mwbranstad, @tjaychen, what is this actually fixing?  Is there something not passing?

**Comment by tjaychen** [comment]
> So @mwbranstad, @tjaychen, what is this actually fixing? Is there something not passing

in the other PR, i was just asking why we have double flop synchronization in a module that's fully synchronous.

**Comment by martin-lueker** [comment]
> 

Got it.  I saw the comment in #14644, but the Async flops are serving a function in this block where they are adding delays, and in the PR I'm actually adding a 3rd cycle delay.  So is this Async option hurting anything?  (Note also that the documentation for the prim_mubi4_sync actually recommends [using Async = 1 when in doubt](https://github.com/lowRISC/opentitan/blob/fc393b28ef56787819d848da95061ac7eabe5661/hw/ip/prim/rtl/prim_mubi4_sync.sv#L24))

> > So @mwbranstad, @tjaychen, what is this actually fixing? Is there something not passing
> 
> in the other PR, i was just asking why we have double flop synchronization in a module that's fully synchronous. It also didn't look like this was coming from another module, but i could be wrong.

Right #14644.   Well we can make explicit delays for all these es_enable_q_fo signals if needed

**Comment by tjaychen** [comment]
> I see. I saw the comment in #14644, but the Async flops are serving a function in this block where they are adding delays, and in the PR I'm actually adding a 3rd cycle delay. So is this Async option hurting anything? (Note also that the documentation for the prim_mubi4_sync actually recommends [using Async = 1 when in doubt](https://github.com/lowRISC/opentitan/blob/fc393b28ef56787819d848da95061ac7eabe5661/hw/ip/prim/rtl/prim_mubi4_sync.sv#L24))
> 
> > > So @mwbranstad, @tjaychen, what is this actually fixing? Is there something not passing
> > 
> > 
> > in the other PR, i was just asking why we have double flop synchronization in a module that's fully synchronous. It also didn't look like this was coming from another module, but i could be wrong.

hmm...if the function is relying on this.. i guess go ahead. It's very odd though that we would need to rely on the delays.
If the other side of the signal were actually async, you wouldn't always get the same timing in reality (so this is kind of alarming).

If the other side were actually sync, it seems like we are using this to time out certain deltas..which I guess would be more okay.

If we definitely need it, I think it would probably be good to just add a comment why the extra delay is needed, otherwise someone else is probably going to come along and eventually try to remove it too.

**Comment by martin-lueker** [comment]
> hmm...if the function is relying on this.. i guess go ahead. It's very odd though that we would need to rely on the delays. If the other side of the signal were actually async, you wouldn't always get the same timing in reality (so this is kind of alarming).

Well this need for delays might be worth discussing more broadly.   The IP itself does not need them per se, (except for making the health_test_clr pulse) but the verification tests get really cranky if the delays are not well understood.   The particular issues are with clearing internal state.   The RNG data obviously does not move through the pipe instantaneously, and the most of the internal state is supposed to be cleared on disable.   Though clearing all this in a way that all the the statistics and health tests and SHA state and output seeds are all consistent, that is a challenge.   

The general approach here is that once RNG data has entered the IP far enough to be included somewhere it shouldn't be cleared until it is included everywhere.  This isn't a spec requirement, but it is done to make verification manageable.

Does that make sense?

> If the other side were actually sync, it seems like we are using this to time out certain deltas..which I guess would be more okay.

Your point is well taken.   This would all fall apart if this were async.  We really can make these explicit.   And then there will be less confusion.    Though can we do this after #14644 lands?   I actually have another PR in the pipe that is built on top of that. 

Furthermore, I just learned this morning that almost nothing is passing the RNG test since #14608 landed, as the original test environment always asserts `cs_aes_halt_ack_i`.   So we need to get a few more fixes in to get everything working before we can test changes like this.  It should be just a couple days though.

**Comment by tjaychen** [comment]
> > hmm...if the function is relying on this.. i guess go ahead. It's very odd though that we would need to rely on the delays. If the other side of the signal were actually async, you wouldn't always get the same timing in reality (so this is kind of alarming).
> 
> Well this need for delays might be worth discussing more broadly. The IP itself does not need them per se, (except for making the health_test_clr pulse) but the verification tests get really cranky if the delays are not well understood. The particular issues are with clearing internal state. The RNG data obviously does not move through the pipe instantaneously, and the most of the internal state is supposed to be cleared on disable. Though clearing all this in a way that all the the statistics and health tests and SHA state and output seeds are all consistent, that is a challenge.
> 
> The general approach here is that once RNG data has entered the IP far enough to be included somewhere it shouldn't be cleared until it is included everywhere. This isn't a spec requirement, but it is done to make verification manageable.
> 
> Does that make sense?
> 
> > If the other side were actually sync, it seems like we are using this to time out certain deltas..which I guess would be more okay.
> 
> Your point is well taken. This would all fall apart if this were async. We really can make these explicit. And then there will be less confusion. Though can we do this after #14644 lands? I actually have another PR in the pipe that is built on top of that.
> 
> Furthermore, I just learned this morning that almost nothing is passing the RNG test since #14608 landed, as the original test environment always asserts `cs_aes_halt_ack_i`. So we need to get a few more fixes in to get everything working before we can test changes like this. It should be just a couple days though.

thanks for clarifying Martin. 
So if there are concerns about this change (and honestly speaking, if we don't change, it's pretty benign), why don't we just hold it for now? I feel like the entropy complex verification is more important than a nice-to-have design cleanup.

**Comment by martin-lueker** [comment]
Well, @mwbranstad, I think it's awesome that you were so responsive in opening this! Though why don't I just put a comment to the effect that the verification is expecting these delays for now.  I can do that in #14644, because I have that open.  Then maybe we can open a P3 issue to clean this up?  What do you all think?

**Comment by martin-lueker** [comment]
Looking at this a but more closely, isn't it possible that the mubi8 OTP signals _are_ asynchronous?

**Comment by martin-lueker** [comment]
My comments above were all about the mubi4 enables.

**Comment by mwbranstad** [comment]
Thanks @tjaychen and @martin-lueker for the discussion. After making this PR, I was starting to slowly remember why these Async On enables exist. Here is what I recall, and what I think should be done:

- mubi4: I believe I purposefully set this to use the built-in delay, merely out of convenience, knowing it was not a true async i/f. This delay was required to sync with other ES logic. Martin mentioned all of this as well. 
- mubi8: both of these signals come from OTP, which I think is a different clock domain. Maybe Tim can confirm. Either way, the safe thing here is to leave it as is.

Bottom line, this PR will be amended to convert actually HW changes to comments, so that we do not run into more 2nd guessing in the future. Sound good?

**Comment by tjaychen** [comment]
that sounds good to me.
